src/components/AccountManagement.vue

```diff
@@ -0,0 +1,161 @@
+  if (!reloginPassword.value) {
+    showRelogin.value = true;
+    return;
+  }
+  // If the password has been entered, log in then delete.
+  try {
+    await signInWithEmailAndPassword(auth, email.value, reloginPassword.value);
+  } catch (error) {
+    // TODO: Show errors in UI.
+    console.log(error);
+    return;
+  }
+  try {
+    await deleteDoc(doc(db, 'users', auth.currentUser!.uid));
+    await deleteUser(auth.currentUser!);
+  } catch (error) {
+    // TODO: Show errors in UI.
+    console.log(error);
+  }
+}
+
+async function updateName() {
+  if (!editingName.value) {
+    editingName.value = true;
+    return;
+  }
+  // TODO: Allow users to save if undefined e.g., no name defined initially.
+  // First, update the user profile in Firebase Auth.
+  try {
+    await updateProfile(auth.currentUser!, {
+      displayName: `${firstName.value} ${lastName.value}`,
+    });
+  } catch (error) {
+    // TODO: Show errors in UI.
+    console.log(error);
+    return;
+  }
+  // Then, also update the Firestore entry for the user.
+  try {
+    await setDoc(doc(db, 'users', auth.currentUser!.uid), {
+      first_name: firstName.value,
+      last_name: lastName.value,
+    });
+  } catch (error) {
+    // TODO: Show errors in UI.
+    console.log(error);
+    return;
+  }
+  editingName.value = false;
+}
+
+// This listener is triggered in two cases:
+// * When the user lands on this page.
+// * When the user deletes their account.
+const unsubscribeAuthListener = onAuthStateChanged(auth, (user) => {
+  if (user) {
+    getUserInfo(user);
+  } else {
+    router.push({ path: '/' });
+  }
+});
+
+onBeforeUnmount(() => {
+  unsubscribeAuthListener();
+});
+</script>
+
+<template>
+  <h2>Account Management</h2>
+  <h3>ID and Password</h3>
+  <div class="email">
+    <label for="email">Email</label>
+    <input name="email" type="text" v-model="email" readonly />
+    <RouterLink to="update-email">Edit</RouterLink>
+  </div>
+  <div class="password">
+    <label for="password">Password</label>
+    <input name="password" type="password" readonly />
+    <RouterLink to="reset-password">Reset</RouterLink>
+  </div>
+  <h3>Personal Information</h3>
+  <form class="name" @submit.prevent>
+    <label for="first-name">Name</label>
+    <input
+      name="first-name"
+      type="text"
+      v-model="firstName"
+      :readonly="!editingName"
+    />
+    <input
+      name="last-name"
+      type="text"
+      v-model="lastName"
+      :readonly="!editingName"
+    />
+    <button @click="updateName" type="submit">
+      {{ editingName ? 'Save' : 'Edit' }}
+    </button>
+  </form>
+  <form class="delete-account" @submit.prevent>
+    <div class="relogin-password" v-if="showRelogin">
+      <label for="password">Password</label>
+      <input
+        name="password"
+        type="password"
+        v-model="reloginPassword"
+        required
+      />
+    </div>
+    <button @click="deleteAccount" type="submit">Delete account</button>
```

Comment:
I know this isn't in the mocks but I think we should also have a logout button on this page, maybe to the left of the delete account button?